core/Plugin/Report.php

```diff
@@ -448,6 +457,38 @@ public function getProcessedMetrics()
         return $this->getMetricTranslations($this->processedMetrics);
     }
 
+    /**
+     * Returns the semantic types for metrics this report displays.
+     *
+     * If the semantic type is not defined by the derived Report class, it defaults to
+     * the value returned by {@link Metrics::getDefaultMetricSemanticTypes()} or
+     * {@link Metric::getSemanticType()}. If the semantic type cannot be found this way,
+     * this method tries to deduce it from the metric name, though this process will
+     * not identify the semantic type for most metrics.
+     *
+     * @return string[] maps metric name => semantic type
+     * @api
+     */
+    public function getMetricSemanticTypes()
```

Comment:
Would be awesome to add (return) type hints to this and all other new methods were possible.
```suggestion
    public function getMetricSemanticTypes(): array
```

Reply:
:+1:

Reply:
@sgiehl parameter type hints as well or just return type hints?

Reply:
parameter type hints as well, as long as they are supported by PHP 7.2 🙂